src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -878,6 +878,11 @@ RangesInDataParts MergeTreeDataSelectExecutor::filterPartsByPrimaryKeyAndSkipInd
                             log);
                     }
 
+                    context->getQueryContext()->addSkipIndexAccessInfo(
+                        ranges.data_part->storage.getStorageID().getFullTableName(),
+                        index_and_condition.index->index.name
+                    );
+
```

Comment:
Let's also add the same for `merged_indices` below, and update test. This is a `hypothesis` index, you can take a look at i.e. `tests/queries/0_stateless/01624_soft_constraints.sh` for example